lib/llm/src/http/service/health.rs

```diff
@@ -0,0 +1,167 @@
+
+    let docs: Vec<RouteDoc> = vec![
+        RouteDoc::new(Method::GET, &path),
+        RouteDoc::new(Method::GET, &path_namespace),
+        RouteDoc::new(Method::GET, &path_component),
+    ];
+
+    let router = Router::new()
+        .route(&path, get(health_handler))
+        .route(&path_namespace, get(health_namespace_handler))
+        .route(&path_component, get(health_component_handler))
+        .with_state(state);
+
+    (docs, router)
+}
+
+async fn health_handler() -> impl IntoResponse {
+    Json(json!({"status": "healthy"}))
+}
+
+// A namespace health check will return if the namespace exists in ETCD and will return a list of components currently registered
+async fn health_namespace_handler(Path(namespace): Path<String>) -> impl IntoResponse {
+    let target_key = format!("{INSTANCE_ROOT_PATH}/{namespace}");
+
+    match get_instances(&target_key).await {
+        Ok(instances) if instances.is_empty() => (
+            StatusCode::NOT_FOUND,
+            Json(json!({"error": "namespace not found", "namespace": namespace})),
+        ),
+        Ok(instances) => {
+            let components: Vec<String> = instances
+                .iter()
+                .map(|i| format!("{}.{}", i.component, i.endpoint))
+                .collect();
+
+            (
+                StatusCode::OK,
+                Json(json!({
+                    "namespace": namespace,
+                    "status": "healthy",
+                    "instance_count": components.len(),
+                    "components": components
+                })),
+            )
+        }
+        Err((status, msg)) => (status, Json(json!({"error": msg}))),
+    }
+}
+
+// A component health check will return the endpoints and instance IDs for a specific component
+async fn health_component_handler(
+    Path((namespace, component)): Path<(String, String)>,
+) -> impl IntoResponse {
+    let target_key = format!("{INSTANCE_ROOT_PATH}/{namespace}/{component}");
+
+    match get_instances(&target_key).await {
+        Ok(instances) if instances.is_empty() => (
+            StatusCode::NOT_FOUND,
+            Json(json!({
+                "error": "component not found",
+                "namespace": namespace,
+                "component": component
+            })),
+        ),
+        Ok(instances) => {
+            let instance_info: Vec<serde_json::Value> = instances
+                .iter()
+                .map(|i| {
+                    json!({
+                        "component": i.component,
+                        "endpoint": i.endpoint,
+                        "instance_id": i.instance_id
+                    })
+                })
+                .collect();
+
+            (
+                StatusCode::OK,
+                Json(json!({
+                    "namespace": namespace,
+                    "component": component,
+                    "status": "healthy",
+                    "instance_count": instances.len(),
```

Comment:
will this double count each endpoint for a component? that is if a component has multiple endpoints does that increase the instance count here?

Reply:
+1